p2p/discover/node.go

```diff
@@ -317,17 +317,18 @@ func PubkeyID(pub *ecdsa.PublicKey) NodeID {
 
 // Pubkey returns the public key represented by the node ID.
 // It returns an error if the ID is not a point on the curve.
-func (id NodeID) Pubkey() (*ecdsa.PublicKey, error) {
+func (n NodeID) Pubkey() (*ecdsa.PublicKey, error) {
 	p := &ecdsa.PublicKey{Curve: crypto.S256(), X: new(big.Int), Y: new(big.Int)}
-	half := len(id) / 2
-	p.X.SetBytes(id[:half])
-	p.Y.SetBytes(id[half:])
+	half := len(n) / 2
+	p.X.SetBytes(n[:half])
+	p.Y.SetBytes(n[half:])
```

Comment:
I don't thin replacing `id` with `n` is better. Please revert this.

Reply:
Ok - I am reverting. The reason I touched this is that it throws the warning: "receiver name id should be consistent with previous receiver name n for NodeID". Close enough to the receiver err I'm working on that I updated. There are many functions in this file that use (n NodeId) - I can either update them all to (id NodeID), or leave them untouched (as I will do after I revert)